src/theory/sets/inference_manager.cpp

```diff
@@ -90,7 +90,7 @@ bool InferenceManager::assertFactRec(Node fact, Node exp, int inferType)
       || (atom.getKind() == EQUAL && atom[0].getType().isSet()))
   {
     // send to equality engine
-    if (assertInternalFact(atom, polarity, InferenceId::UNKNOWN, exp))
+    if (assertInternalFact(atom, polarity, InferenceId::SETS_EQUALITY, exp))
```

Comment:
Shouldn't member and equal have different Ids here?

Reply:
Possibly, should they just be `SETS_EQUALITY` and `SETS_MEMBER`?

Reply:
Actually, this is another case where this method (assertFactRec) should take an `InferenceId`: this is a utility method used by many different inference schemas in sets.

Reply:
This one (lnie 93) still needs to change to id